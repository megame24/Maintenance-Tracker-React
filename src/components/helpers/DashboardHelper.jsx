import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

export const TitleColumn = ({ request }) => (
  <td className="dark-blue table-link">
    <Link to={`/admin/requests/${request.id}`}>
      {request.title}
    </Link>
  </td>
);

TitleColumn.propTypes = {
  request: PropTypes.shape({
    title: PropTypes.string
  }).isRequired
};

export const ApproveAndDisapproveColumn = ({
  request, approveRequest, disapproveRequest
}) => (
  <td>
    {request.status === 'pending' && (
      <span>
        <a
          onClick={() => approveRequest(request.id)}
          className="approve btn btn-small btn-primary"
        >
          Approve
        </a>
        &nbsp;&nbsp;&nbsp;
        <a
          onClick={() => disapproveRequest(request.id)}
          className="disapprove
          btn btn-small btn-danger"
        >
          Disapprove
        </a>
      </span>
    )}
    {(request.status === 'approved'
      || request.status === 'resolved') && 'Approved'}
    {request.status === 'disapproved' && 'Disapproved'}
  </td>
);

ApproveAndDisapproveColumn.propTypes = {
  request: PropTypes.shape({
    title: PropTypes.string
  }).isRequired,
  approveRequest: PropTypes.func.isRequired,
  disapproveRequest: PropTypes.func.isRequired,
};

export const ResolveColumn = ({ request, resolveRequest }) => (
  <td>
    {request.status === 'approved' && (
      <a
        onClick={() => resolveRequest(request.id)}
        className="resolve btn btn-small btn-success"
      >
        Resolve
      </a>
    )}
    {request.status === 'resolved' && 'Resolved'}
  </td>
);

ResolveColumn.propTypes = {
  request: PropTypes.shape({
    title: PropTypes.string
  }).isRequired,
  resolveRequest: PropTypes.func.isRequired,
};

export const Table = ({
  requests, approveRequest, disapproveRequest, resolveRequest,
}) => (
  <table>
    <thead>
      <tr>
        <th>Title</th>
        <th className="table-not-mobile">Type</th>
        <th>Approve / Disapprove</th>
        <th>Resolve</th>
      </tr>
    </thead>
    <tbody id="table-body">
      {requests
        .map(request => (
          <tr key={request.id}>
            <TitleColumn request={request} />
            <td className="table-not-mobile">{request.type}</td>
            <ApproveAndDisapproveColumn
              request={request}
              approveRequest={approveRequest}
              disapproveRequest={disapproveRequest}
            />
            <ResolveColumn
              request={request}
              resolveRequest={resolveRequest}
            />
          </tr>
        ))}
    </tbody>
  </table>
);

Table.propTypes = {
  requests: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string
  })),
  approveRequest: PropTypes.func.isRequired,
  disapproveRequest: PropTypes.func.isRequired,
  resolveRequest: PropTypes.func.isRequired,
};

Table.defaultProps = {
  requests: {}
};

export default {};
