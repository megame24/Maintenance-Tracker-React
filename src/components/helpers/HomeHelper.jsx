import React from 'react';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';

export const Banner = ({ token }) => (
  <div className="hero">
    <div className="text container">
      <h1>
        Use,
        <br />
        Maintain,
        <br />
        Repair,
        <br />
        Repeat.
        <br />
      </h1>
      <p>
        We take care of your maintenance and
        repair hassles, so you don't have to.
      </p>
      <Link
        to={token ? '/admin/dashboard' : '/signup'}
        className="btn btn-tertiary get-started"
      >
          Get started
      </Link>
    </div>
  </div>
);

Banner.propTypes = {
  token: PropTypes.string,
};

Banner.defaultProps = {
  token: ''
};

export const HomeBody = () => (
  <div className="how-it-works center">
    <h2>How It Works</h2>
    <div className="flex">
      <div>
        <i className="icon ion-paper-airplane" />
        <h4>Make a request</h4>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit,
          sed do eiusmod tempor incididunt ut labore
          et dolore magna aliqua. Ut enim ad minim veniam.
        </p>
      </div>
      <div>
        <i className="icon ion-leaf" />
        <h4>Request accepted</h4>
        <p>
          Lorem ipsum dolor sit amet, consectetur elit,
          sed do eiusmod tempor incididunt ut labore et dolore
          magna aliqua. Ut enim ad minim veniam.
        </p>
      </div>
      <div>
        <i className="icon ion-ios-bolt" />
        <h4>Request resolved</h4>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit,
          sed do eiusmod tempor incididunt ut labore
          et dolore magna veniam.
        </p>
      </div>
    </div>
  </div>
);

export const HomeFooter = ({ token }) => (
  <p className="center">
    <Link
      to={token ? '/admin/dashboard' : '/signup'}
      className="btn btn-small btn-primary get-started"
    >
      Click here
    </Link>
    &nbsp;
    to get started
  </p>
);

HomeFooter.propTypes = {
  token: PropTypes.string,
};

HomeFooter.defaultProps = {
  token: ''
};

export default {};
