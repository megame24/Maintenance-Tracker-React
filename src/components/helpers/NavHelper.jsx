import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

export const UsersNav = ({
  notMobile, dropDown, toggleDropDown, username, logout
}) => (
  <ul className={`right-menu ${notMobile} not-mobile-right`}>
    <li key="request" className="request">
      <div>
        <a
          onClick={() => toggleDropDown('request')}
          href="#"
          className="nav-item"
        >
          Requests
          <span>
            <i className="icon ion-android-arrow-dropdown" />
          </span>
        </a>
      </div>
      <div className={`drop-down-menu ${dropDown.request}`}>
        <Link to="/make-request">Make Request</Link>
        <Link to="/view-requests">View Requests</Link>
      </div>
    </li>
    <li key="user" className="user">
      <div>
        <a
          href="#"
          onClick={() => toggleDropDown('users')}
          className="nav-item"
        >
          <span id="display-username">
            Welcome
            &nbsp;
            {username}
          </span>
          <span>
            <i className="icon ion-android-arrow-dropdown" />
          </span>
        </a>
      </div>
      <div className={`drop-down-menu ${dropDown.users}`}>
        <a id="logout" href="#" onClick={logout}>Logout</a>
      </div>
    </li>
  </ul>
);

UsersNav.propTypes = {
  notMobile: PropTypes.string,
  dropDown: PropTypes.shape({}).isRequired,
  toggleDropDown: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired
};

UsersNav.defaultProps = {
  notMobile: '',
};

export const AdminNav = ({
  notMobile, username, dropDown, toggleDropDown, logout
}) => (
  <ul className={`right-menu ${notMobile} not-mobile-right`}>
    <li className="admin">
      <div>
        <a
          onClick={() => toggleDropDown('admin')}
          href="#"
          className="nav-item"
        >
          <span id="display-username">
            Welcome
            &nbsp;
            {username}
          </span>
          <span>
            <i className="icon ion-android-arrow-dropdown" />
          </span>
        </a>
      </div>
      <div className={`drop-down-menu ${dropDown.admin}`}>
        <Link to="/dashboard">Dashboard</Link>
        <a id="logout" href="#" onClick={logout}>Logout</a>
      </div>
    </li>
  </ul>
);

AdminNav.propTypes = {
  notMobile: PropTypes.string,
  dropDown: PropTypes.shape({}).isRequired,
  toggleDropDown: PropTypes.func.isRequired,
  logout: PropTypes.func.isRequired,
  username: PropTypes.string.isRequired
};

AdminNav.defaultProps = {
  notMobile: '',
};

export const GuestNav = ({
  notMobile
}) => (
  <ul className={`right-menu ${notMobile} not-mobile-right`}>
    <li>
      <Link to="/login" className="nav-item">Login</Link>
    </li>
    <li>
      <Link to="/signup" className="nav-item">Sign Up</Link>
    </li>
  </ul>
);

GuestNav.propTypes = {
  notMobile: PropTypes.string,
};

GuestNav.defaultProps = {
  notMobile: '',
};

export const NavHeader = ({
  navHeaderShadow, toggleMenu,
}) => (
  <div className={`nav-header ${navHeaderShadow} not-mobile-left`}>
    <Link to="/">M-Tracker</Link>
    <span className="mobile toggle-menu" onClick={toggleMenu}>
      <i className="icon ion-android-menu" />
    </span>
    <div className="clearfix" />
  </div>
);

NavHeader.propTypes = {
  navHeaderShadow: PropTypes.string,
  toggleMenu: PropTypes.func.isRequired,
};

NavHeader.defaultProps = {
  navHeaderShadow: '',
};

export default {};
