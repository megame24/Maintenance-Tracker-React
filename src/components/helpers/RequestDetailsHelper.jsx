import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

export const RequestDetail = ({ request, statusColor, statusMessage }) => (
  <div className="request-detail">
    <h2>{request.title}</h2>
    <p>{request.description}</p>
    <p>
      <strong>Request date:</strong>
      &nbsp;
      {new Date(Number(request.date)).toDateString()}
    </p>
    <div
      className={`status ${statusColor[request.status]}`}
    >
      {statusMessage[request.status]}
    </div>
    <div className="request-type">{request.type}</div>
  </div>
);

RequestDetail.propTypes = {
  statusColor: PropTypes.shape({}).isRequired,
  statusMessage: PropTypes.shape({}).isRequired,
  request: PropTypes.shape({
    title: PropTypes.string
  }).isRequired,
};

export const AdminFeedbackAndCtrlBtns = ({ request }) => (
  <div>
    <h4>Feedback from an admin</h4>
    <div className="feedback">
      <p>{request.feedback}</p>
    </div>
    {request.status === 'pending' && (
      <span>
        <Link
          to={`/users/requests/edit/${request.id}`}
          className="btn btn-tertiary"
        >
          Edit&nbsp;
          <i className="icon ion-edit" />
        </Link>
        &nbsp;&nbsp;
      </span>
    )}
    {
      (request.status === 'resolved'
        || request.status === 'disapproved'
        || request.status === 'pending')
      && (
        <a
          id="delete"
          className="btn btn-danger"
        >
          Delete&nbsp;
          <i className="icon ion-android-delete" />
        </a>
      )
    }
  </div>
);

AdminFeedbackAndCtrlBtns.propTypes = {
  request: PropTypes.shape({
    title: PropTypes.string
  }).isRequired,
};

export default {};
