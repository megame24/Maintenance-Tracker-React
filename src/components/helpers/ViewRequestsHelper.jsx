import React from 'react';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

export const TableRow = ({ request, statusColor, statusMessage }) => (
  <tr>
    <td className="dark-blue table-link">
      <Link to={`/users/requests/${request.id}`}>
        {request.title}
      </Link>
    </td>
    <td>{request.type}</td>
    <td
      className={statusColor[request.status]}
    >
      {statusMessage[request.status]}
    </td>
  </tr>
);

TableRow.propTypes = {
  statusColor: PropTypes.shape({}).isRequired,
  statusMessage: PropTypes.shape({}).isRequired,
  request: PropTypes.shape({
    title: PropTypes.string
  }).isRequired,
};

export const Table = ({ requests, statusColor, statusMessage }) => (
  <table>
    <thead>
      <tr>
        <th>Title</th>
        <th>Type</th>
        <th>Status</th>
      </tr>
    </thead>
    <tbody id="table-body">
      {
        requests
          .map(request => (
            <TableRow
              key={request.id}
              request={request}
              statusColor={statusColor}
              statusMessage={statusMessage}
            />
          ))
      }
    </tbody>
  </table>
);

Table.propTypes = {
  statusColor: PropTypes.shape({}).isRequired,
  statusMessage: PropTypes.shape({}).isRequired,
  requests: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string
  })),
};

Table.defaultProps = {
  requests: [],
};

export const NoRequests = () => (
  <div className="no-requests">
    <i className="ion-plus-circled icon" />
    <br />
    <br />
    <p>
      You have no requests at the moment,
      <Link
        to="/users/requests/create"
        className="btn btn-primary btn-small"
      >
        click here
      </Link>
      to make a request
    </p>
  </div>
);

export default {};
