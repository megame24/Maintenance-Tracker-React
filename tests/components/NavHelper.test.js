import React from 'react';
import { shallow } from "enzyme";
import { UsersNav, AdminNav, GuestNav, NavHeader } from '../../src/components/helpers/NavHelper';

describe('Testing UsersNav component', () => {
  it('should render as expected', () => {
    const props = {
      logout: () => {},
      notMobile: 'not-mobile',
      dropDown: {},
      toggleDropDown: () => {},
      username: 'username',
    }
    const tree = shallow(<UsersNav {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing AdminNav component', () => {
  it('should render as expected', () => {
    const props = {
      logout: () => {},
      notMobile: 'not-mobile',
      dropDown: {},
      toggleDropDown: () => {},
      username: 'username',
    }
    const tree = shallow(<AdminNav {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing GuestNav component', () => {
  it('should render as expected', () => {
    const tree = shallow(<GuestNav />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing NavHeader component', () => {
  it('should render as expected', () => {
    const props = {
      navHeaderShadow: 'nav-header-shadow',
      toggleMenu: () => {},
    }
    const tree = shallow(<NavHeader {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
