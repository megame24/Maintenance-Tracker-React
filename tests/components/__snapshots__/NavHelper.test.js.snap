// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AdminNav component should render as expected 1`] = `
<ul
  className="right-menu not-mobile not-mobile-right"
>
  <li
    className="admin"
  >
    <div>
      <a
        className="nav-item"
        onClick={[Function]}
      >
        <span
          id="display-username"
        >
          Welcome  
          username
        </span>
        <span>
          <i
            className="icon ion-android-arrow-dropdown"
          />
        </span>
      </a>
    </div>
    <div
      className="drop-down-menu undefined"
    >
      <Link
        replace={false}
        to="/dashboard"
      >
        Dashboard
      </Link>
      <a
        id="logout"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
  </li>
</ul>
`;

exports[`Testing GuestNav component should render as expected 1`] = `
<ul
  className="right-menu  not-mobile-right"
>
  <li>
    <Link
      className="nav-item"
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </li>
  <li>
    <Link
      className="nav-item"
      replace={false}
      to="/signup"
    >
      Sign Up
    </Link>
  </li>
</ul>
`;

exports[`Testing NavHeader component should render as expected 1`] = `
<div
  className="nav-header nav-header-shadow not-mobile-left"
>
  <Link
    replace={false}
    to="/"
  >
    M-Tracker
  </Link>
  <span
    className="mobile toggle-menu"
    onClick={[Function]}
  >
    <i
      className="icon ion-android-menu"
    />
  </span>
  <div
    className="clearfix"
  />
</div>
`;

exports[`Testing UsersNav component should render as expected 1`] = `
<ul
  className="right-menu not-mobile not-mobile-right"
>
  <li
    className="request"
    key="request"
  >
    <div>
      <a
        className="nav-item"
        onClick={[Function]}
      >
        Requests
        <span>
          <i
            className="icon ion-android-arrow-dropdown"
          />
        </span>
      </a>
    </div>
    <div
      className="drop-down-menu undefined"
    >
      <Link
        replace={false}
        to="/users/requests/create"
      >
        Create Request
      </Link>
      <Link
        replace={false}
        to="/users/requests/all"
      >
        View Requests
      </Link>
    </div>
  </li>
  <li
    className="user"
    key="user"
  >
    <div>
      <a
        className="nav-item"
        onClick={[Function]}
      >
        <span
          id="display-username"
        >
          Welcome  
          username
        </span>
        <span>
          <i
            className="icon ion-android-arrow-dropdown"
          />
        </span>
      </a>
    </div>
    <div
      className="drop-down-menu undefined"
    >
      <a
        id="logout"
        onClick={[Function]}
      >
        Logout
      </a>
    </div>
  </li>
</ul>
`;
